fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,297 @@
+
+    kind : str
+        The type of plot to display. e.g. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    legend_label : str
+        The label corresponding to the confidence interval bars
+    """
+    color = ax.lines[0].get_color() if kind == "scatter" else "black"
+
+    # extend legend with user-provided legend_label
+    handles, labels = ax.get_legend_handles_labels()
+    custom_line = [Line2D([0], [0], color=color, label=legend_label)]
+    handles.extend(custom_line)
+    ax.legend(handles=handles)
+
+
+def _plot_df(
+    df, metrics, kind, subplots, legend_label, df_all_errors=None, **kwargs
+):
+    r"""Plot the data with or without errors.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The collection of disaggregated metric values, along with the metric errors.
+
+    metrics : str or list of str
+        The name of the metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+    kind : str
+        The type of plot to display. e.g. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    subplots : bool
+        Whether or not to plot metrics on separate subplots
+
+    legend_label : str
+        The label corresponding to the confidence interval bars
+
+    df_all_errors : pd.DataFrame, default=None
+        Optional argument for DataFrame that indicates the errors to plot
+
+    \*\*kwargs
+        Keyword arguments that are passed in to :meth:`pandas.DataFrame.plot`.
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    if df_all_errors is not None:
+        yerr = np.array(
+            [
+                np.array([[*row] for row in df_all_errors[metric]]).T
+                for metric in metrics
+            ]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="",
+                marker="o",
+                yerr=yerr,
+                subplots=subplots,
+                **kwargs
+            )
+        else:
+            axs = df[metrics].plot(
+                kind=kind, yerr=yerr, subplots=subplots, **kwargs
+            )
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                _build_legend(ax, kind, legend_label)
+        else:
+            _build_legend(axs, kind, legend_label)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="", marker="o", subplots=subplots, **kwargs
+            )
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(
+    metric_frame: MetricFrame,
+    *,
+    kind: str = "scatter",
+    metrics: Union[List[str], str] = None,
+    conf_intervals: Union[List[str], str] = None,
+    subplots: bool = True,
+    plot_error_labels: bool = False,
```

Comment:
And what are the thoughts on renaming all the arguments `plot_error_labels` and `error_*` to something like `plot_ci_labels` and `ci_*`? Our choice to use "error" in argument names dates back to when the argument that we were passing for the error bars was called `errors` or something like that but now it is `conf_intervals`.

Reply:
Done. And since there's a `*` in the argument list, this is relatively easy to change in future if we want.

Reply:
@riedgar-ms : we also need to change `plot_error_labels` -> `plot_ci_labels`